pallets/exchange/src/direct.rs

```diff
@@ -28,6 +28,8 @@ impl<'a, T: Config> DirectTradeData<'a, T> {
 	/// 1. Validate balances
 	/// 2. Calculate fees
 	/// 3. Reserve amounts for each transfer ( including fee transfers )
+	// REVIEW: I would investigate using storage transactions instead of keeping track of all the
+	// transfers here.
```

Comment:
We will definitely look at the atomic and use of storage transactions as suggested. However, I feel this would require lot of effort and refactor to change the exchange pallet and algorithm itself. So we will focus on thorough testing now and keeping this in mind when we will come back to improving the exchange pallet and algorithm ( track issue #94 )3

Reply:
I agree that you probably want to look this over and think about it. You can already make `execute_buy` and `execute_sell` transactional, though.